quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -235,9 +239,18 @@ public Optional<DeprecatedMetadata> getDeprecatedMetadata() {
     private ConfigValue transformValue(String name, String value, ConfigSourceInterceptorContext context, String configSourceName, boolean parentValue) {
         String mappedValue = value;
         
+        boolean mapped = false;
         var theMapper = parentValue ? this.parentMapper : this.mapper;
         if (theMapper != null && (!name.equals(getFrom()) || parentValue)) {
             mappedValue = theMapper.apply(value, context);
+            mapped = true;
+        }
+        
+        // defaults and values from transformers may not have been subject to expansion
+        if ((mapped || configSourceName == null) && mappedValue != null && Expressions.isEnabled() && mappedValue.contains("$")) {
+            mappedValue = new ExpressionConfigSourceInterceptor().getValue(
```

Comment:
The usage of `SystemPropertyReplacer` is also able to return a default value in a placeholder when the property is not set. For instance, `${name:default-value}`.

Is it still the case? Not sure if we have coverage for that in our dist testsuite or perhaps we do indirectly when running the base testsuite.

Reply:
Quarkus/smallrye supports the same syntax. It is in use in our database urls, and we do have coverage there in ConfigurationTest.

Reply:
@pedroigor Do the changes look good to you with what @shawkins replied above?